Civi/Share/ChangeProcessingEvent.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace Civi\Share;
+
+use Civi\Core\Event\GenericHookEvent as Event;
+use Civi\Api4\ShareChange;
+use Civi\Api4\ShareNode;
+
+/**
+ * CiviShare ChangeProcessingEvent
+ *
+ * The purpose of this class is to process/apply a set of changes
+ *  to the system.
+ *
+ * It will trigger a symfony hook to determine which processor instances would
+ *  want to process the change at hand an will make them apply it.
+ *
+ * Processors can have a look a the set of changes to determine whether they
+ *  can process it.
+ *
+ * Provided by the CiviShare extension.
+ */
+class ChangeProcessingEvent extends Event
+{
+
+  /** @var int node ID */
+  protected int $node_id;
+
+  /** @var int change ID */
+  protected int $change_id;
+
+  /** @var array node data */
+  protected ?array $node_data = null;
+
+  /** @var array change data */
+  protected ?array $change_data = null;
```

Comment:
This could be a separate class `Change` (or maybe two for changes to be sent and received changes) with defined properties and methods for things currently done by the `Message` class, e.g. `markAllChanges()`, `processChanges()`, `setChangeStatus()`, etc.

Reply:
Disagree, but let's discuss